runtime/lua/vim/lsp.lua

```diff
@@ -25,6 +25,27 @@ local lsp = {
   -- format_rpc_error = lsp_rpc.format_rpc_error;
 }
 
+-- maps request name to the required resolved_capability in the client.
+local req_to_capability = {
```

Comment:
I think maybe we should expose this as `vim.lsp._request_name_to_capability` -- so people can add different methods to check if they need to later (and not just to default to true).

Reply:
Perhaps we can ship as-is and see if there's demand for exposing it? I'm assuming that removing something from the public API is harder than adding it 🙈 (or does the `_` prefix gives us some freedom to remove it later if we figure it's not used).

Reply:
`_` prefix means "use at your own risk".

Reply:
(so we can change whenever and however we want, particularly before we release)

At this point, if we have a good reason, we can break "compatibility" with earlier 0.5 items because we haven't released 0.5 yet anyway (even for non "private" items). That is what happens when you follow master sometimes (rarely, but it does happen)

Reply:
Sounds good, will adjust then!

Reply:
Done! 🎉